R/map.R

```diff
@@ -160,3 +175,21 @@ walk <- function(.x, .f, ..., .progress = FALSE) {
   map(.x, .f, ..., .progress = .progress)
   invisible(.x)
 }
+
+with_indexed_errors <- function(expr, i, error_call = caller_env()) {
+  withCallingHandlers(
+    expr,
+    error = function(cnd) {
+      if (i == 0L) {
+        # error happened before or after loop
+        cnd_signal(cnd)
+      } else {
+        cli::cli_abort(
+          "Can't compute index {i}.",
```

Comment:
For instance we'd have:

```r
f <- function(x) x + 1

tibble(z = 1:2) %>%
  group_by(z) %>%
  summarise(y = f(""))
#> Error in `summarise()`:
#> ℹ In argument `y = f("")`.
#> ℹ In group 1: `z = 1`.
#> Caused by error in `x + 1`:
#> ! non-numeric argument to binary operator
```

Instead of:

```r
#> Error in `summarise()`:
#> ! Can't compute `y = f("")`.
#> ℹ In group 1: `z = 1`.
#> Caused by error in `x + 1`:
#> ! non-numeric argument to binary operator
```

By contrast, in this case, we'd keep the "Can't" form because an error actually happened at the contextual level:

```r
tibble(z = 1:2) %>%
  group_by(z) %>%
  summarise(x = 1:3, y = 1:2)
#> Error in `summarise()`:
#> ! Can't recycle `y = 1:2`.
#> ℹ In group 1: `z = 1`.
#> Caused by error:
#> ! `y` must be size 3 or 1, not 2.
#> ℹ An earlier column had size 3.
```

Reply:
I like that `In` is for "informational" bullets (i.e. the ones that use the `i` bullet) and `Can't` is for error bullets that are prefixed with `!` or `x`

Reply:
What do you think about transforming the "! Can't compute" bullets in "i In argument" bullets?

I think I'll open a dplyr PR with this change to discuss the proposal.